sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/TableManager.java

```diff
@@ -0,0 +1,296 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.schema;
+
+import io.opentelemetry.extension.annotations.WithSpan;
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.bridge.proto.StargateBridge;
+import io.stargate.sgv2.docsapi.api.common.StargateRequestInfo;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentTableColumns;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentTableProperties;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCode;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCodeRuntimeException;
+import io.stargate.sgv2.docsapi.service.schema.common.SchemaManager;
+import io.stargate.sgv2.docsapi.service.schema.query.CollectionQueryProvider;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+/** Table manager provides basic operations on tables that are used for storing collections. */
+@ApplicationScoped
+public class TableManager {
+
+  @Inject DocumentProperties documentProperties;
+
+  @Inject SchemaManager schemaManager;
+
+  @Inject CollectionQueryProvider collectionQueryProvider;
+
+  @Inject StargateRequestInfo requestInfo;
+
+  /**
+   * Fetches a table from the schema manager. Subclasses can override to use the authorized version.
+   *
+   * @param keyspaceName Keyspace
+   * @param tableName Table
+   * @return Table from schema manager
+   */
+  protected Uni<Schema.CqlTable> getTable(String keyspaceName, String tableName) {
+    return schemaManager.getTable(keyspaceName, tableName, getMissingKeyspaceFailure(keyspaceName));
+  }
+
+  /**
+   * Supplier for the correct failure in case of the missing keyspace.
+   *
+   * @param keyspaceName Keyspace
+   * @return Uni emitting failure with {@link ErrorCode#DATASTORE_KEYSPACE_DOES_NOT_EXIST}.
+   */
+  protected Supplier<Uni<? extends Schema.CqlKeyspaceDescribe>> getMissingKeyspaceFailure(
+      String keyspaceName) {
+    return () -> {
+      String message = "Unknown namespace %s, you must create it first.".formatted(keyspaceName);
+      Exception exception =
+          new ErrorCodeRuntimeException(ErrorCode.DATASTORE_KEYSPACE_DOES_NOT_EXIST, message);
+      return Uni.createFrom().failure(exception);
+    };
```

Comment:
It feels a bit wasteful to create this supplier every time.
Could `SchemaManager` take a `Function<String, Uni<...>>` instead? Then we could cache it as a constant.